include/deal.II/fe/fe_values.h

```diff
@@ -2175,9 +2177,24 @@ class FEValuesBase : public Subscriptor
   static const unsigned int space_dimension = spacedim;
 
   /**
-   * Number of quadrature points.
+   * Number of quadrature points of the current object. Its value is
+   * initialized by the value of max_quadrature_points and is updated,
+   * e.g., if FEFaceValues::reinit() is called for a new cell/face.
+   *
+   * @note The default value equals to the value of max_quadrature_points.
    */
-  const unsigned int n_quadrature_points;
+  unsigned int n_quadrature_points;
+
+  /**
+   * Maximum number of quadrature points. This value might be different from
+   * n_quadrature_points, e.g., if a QCollection with different face quadrature
+   * rules has been passed to initialize FEFaceValues.
+   *
+   * This is mostly useful to initialize arrays to allocate the maximum amount
+   * of memory that may be used when re-sizing later on to a the current
+   * number of quadrature points given by n_quadrature_points.
+   */
+  const unsigned int max_quadrature_points;
```

Comment:
I think we more often use the pattern `max_n_quadrature_points` for this kind of names.